### PR TITLE
Move AWS SPI code into Pekko repo

### PR DESCRIPTION
**Pull request by pjfanning**
* See https://github.com/apache/pekko-connectors/issues/612
* follow up in https://github.com/apache/pekko-connectors/issues/655

### CODE REVIEW

**Comment by pjfanning** [comment]
@mdedetrich I'm thinking of just removing the integration tests. They require real AWS access. There are unit tests that use testcontainers. I think those are enough. Is this ok?

**Comment by mdedetrich** [comment]
> @mdedetrich I'm thinking of just removing the integration tests. They require real AWS access. There are unit tests that use testcontainers. I think those are enough. Is this ok?

So we do actually have a real AWS account that is provided by Apache INFRA and it is currently being used by [S3 tests](https://github.com/apache/pekko-connectors/blob/8c5d65051cfeb3df4a0eb87039c33131b6b21ccc/s3/src/test/scala/org/apache/pekko/stream/connectors/s3/scaladsl/S3IntegrationSpec.scala#L1267-L1282) and there is a [github action that triggers it daily off the main branch](https://github.com/apache/pekko-connectors/blob/e77bc690d85e91134c5a9571d733ea2a66c3de67/.github/workflows/nightly-builds.yaml)

Initially I would suggest trying to enable the tests and if its too much effort we can disable them, file an issue and try and re-enable them later. The reason I am suggesting this is that the core reason for me pushing for the AWS S3 integration tests is we did have an actual case of a regression that wasn't caught since initially there wasn't tests running against it.

**Comment by mdedetrich** [comment]
Oh and I forgot to mention that if you do end up enabling integration tests, you have to work with Apache INFRA so that they can add in the necessary roles/permissions in the AWS account for whatever is being tested.

This does require some back and forth so given that it might make sense to do it in a different PR, but regardless I think we should aim to have these tests in place

**Comment by mdedetrich** [comment]
@pjfanning Is this ready

**Comment by pjfanning** [comment]
I'll leave the integration tests to another time but I have a couple of more things to do with this PR - it is pretty close now.

File: aws-spi-pekko-http/src/it/scala/org/apache/pekko/stream/connectors/awsspi/dynamodb/ITTestDynamoDB.scala [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+import pekko.stream.connectors.awsspi.{ PekkoHttpAsyncHttpService, TestBase }
+import pekko.util.FutureConverters
+import org.scalatest.concurrent.{ Eventually, Futures, IntegrationPatience }
+import org.scalatest.concurrent.ScalaFutures._
+import org.scalatest.wordspec.AnyWordSpec
+import org.scalatest.matchers.should.Matchers
+import software.amazon.awssdk.services.dynamodb.DynamoDbAsyncClient
+import software.amazon.awssdk.services.dynamodb.model._
+
+class ITTestDynamoDB
```
Comment (laglangyue) [inline aws-spi-pekko-http/src/it/scala/org/apache/pekko/stream/connectors/awsspi/dynamodb/ITTestDynamoDB.scala:30]: It seems that `IT` and `Test` are usually placed at the end

**Comment by He-Pin** [comment]
I will take a look this weekend

File: build.sbt [hunk 2]
```diff
@@ -128,7 +129,13 @@ lazy val amqp = pekkoConnectorProject("amqp", "amqp", Dependencies.Amqp)
 lazy val avroparquet =
   pekkoConnectorProject("avroparquet", "avroparquet", Dependencies.AvroParquet)
 
-lazy val awslambda = pekkoConnectorProject("awslambda", "aws.lambda", Dependencies.AwsLambda)
+lazy val awsSpiPekkoHttp =
+  pekkoConnectorProject("aws-spi-pekko-http", "aws.api.pekko.http", Dependencies.AwsSpiPekkoHttp)
+    .configs(IntegrationTest)
+    .settings(Defaults.itSettings)
+
+lazy val awslambda = pekkoConnectorProject("awslambda", "aws.api.pekko.http", Dependencies.AwsLambda)
```
Comment (samueleresca) [inline build.sbt:137]: Is the moduleName correct? Shoudln't be:

`pekkoConnectorProject("awslambda", "aws.lambda", Dependencies.AwsLambda)`

Comment (pjfanning) [reply build.sbt]: thanks - I've fixed the module name now

**Review by samueleresca — COMMENTED** [review]
looks good to me, Can someone else add their review?

**Comment by pjfanning** [comment]
Would anyone have time to review this PR? I'm hoping to finalise the 1.1.0-M1 RC1 over the next week or so.

File: aws-spi-pekko-http/src/it/scala/org/apache/pekko/stream/connectors/awsspi/dynamodb/ITTestDynamoDB.scala (hunk 1 above)
Comment (nvollmar) [reply aws-spi-pekko-http/src/it/scala/org/apache/pekko/stream/connectors/awsspi/dynamodb/ITTestDynamoDB.scala:30]: Agree, should be a post fix

**Review by nvollmar — APPROVED** [review]
lgtm
Just test naming should be fixed at some point

**Review by samueleresca — APPROVED** [review]
already reviewed. I'm proceeding by approving.

**Review by mdedetrich — APPROVED** [review]
lgtm, sorry for taking a while

**Pull request merged**